src/ArcGISRuntime.Samples.Shared/Managers/ApiKeyManager.cs

```
@@ -0,0 +1,206 @@
+
+using Map = Esri.ArcGISRuntime.Mapping.Map;
+
+namespace ArcGISRuntime.Samples.Shared.Managers
+{
+    public static class ApiKeyManager
+    {
+        private static string _key;
+        private const string _apiKeyFileName = "agolResource";
+
+        public static string ArcGISDeveloperApiKey
+        {
+            get
+            {
+                // An Application programming interface key (API key) is a unique identifier used to authenticate a user, developer, or calling program with a server portal.
+                // Typically, API keys are used to authenticate a calling program within the API rather than an individual user.
+                // Go to https://links.esri.com/arcgis-runtime-security-auth to learn how to obtain a developer API key for ArcGIS Online.
+
+                return _key;
+            }
+
+            set
+            {
+                Esri.ArcGISRuntime.ArcGISRuntimeEnvironment.ApiKey = _key = value;
+            }
+        }
+
+        public async static Task<ApiKeyStatus> CheckKeyValidity()
+        {
+            // Check that a key has been set.
+            if (string.IsNullOrEmpty(Esri.ArcGISRuntime.ArcGISRuntimeEnvironment.ApiKey)) return ApiKeyStatus.Missing;
+
+            // Check that key is valid for loading a basemap.
+            try
+            {
+                await new Map(BasemapStyle.ArcGISTopographic).LoadAsync();
+                return ApiKeyStatus.Valid;
+            }
+            catch (Exception)
+            {
+                return ApiKeyStatus.Invalid;
+            }
+        }
+
+        public static async Task<bool> TrySetLocalKey()
+        {
+            // Check for a local key if a key is not already set.
+            if (ApiKeyManager.ArcGISDeveloperApiKey == null)
+            {
+                try
+                {
+                    ApiKeyManager.ArcGISDeveloperApiKey = await GetLocalKey();
+                    return true;
+                }
+                catch (Exception)
```

Is this going to swallow any exceptions? Maybe better to handle or rethrow?

---

The intention is to catch the HTTP error when an invalid key is used to load a map with the new basemap style. An HTTP exception means the key did not work, or there was no internet connection. Returning an `ApiKeyStatus.Invalid` instead of throwing the exception further is by design.